src/mawpy/steps/incremental_clustering.py

```diff
@@ -0,0 +1,396 @@
+"""
+Perform clustering on multiple locations of one user based on a spatial threshold
+Each cluster of locations represents a potential stay
+
+input:
+    gps stay information / celluar stay information
+    spatial threshold
+    duration constraint threshold (for detect common stay)
+output:
+    potential stays represented by clusters of locations
+"""
+
+import sys
+from datetime import datetime
+
+import psutil
+import time
+import numpy as np
+import pandas as pd
+
+from multiprocessing import Pool
+from multiprocessing import Lock, cpu_count
+
+from geopy.distance import distance
+from sklearn.cluster import KMeans
+
+from mawpy.cluster import Cluster
+from mawpy.constants import USER_ID, STAY_DUR, ORIG_LAT, STAY_LAT, STAY_LONG, UNIX_START_T, UNIX_START_DATE, STAY_UNC, \
+    ORIG_LONG, ORIG_UNC, STAY, STAY_LAT_PRE_COMBINED, STAY_LONG_PRE_COMBINED, STAY_PRE_COMBINED
+from mawpy.utilities.preprocessing import get_preprocessed_dataframe, get_list_of_chunks_by_column
+
+
+def init(this_lock):
+    global lock
+    lock = this_lock
+
+
+def _get_cluster_center(row, mapping, dur_constr):
+    """
+    Returns the lat and long of the cluster to which the trace(row) belongs.
+    """
+    if dur_constr:
+        lat_long = (str(row[STAY_LAT]), str(row[STAY_LONG]))
+        unc = row[STAY_UNC]
+    else:
+        lat_long = (str(row[ORIG_LAT]), str(row[ORIG_LONG]))
+        unc = row[ORIG_UNC]
+
+    if lat_long in mapping:
+        cluster_lat, cluster_long, cluster_radius = mapping[lat_long]
+        return cluster_lat, cluster_long, max(unc, cluster_radius)
+    else:
+        return lat_long[0], lat_long[1], unc
+
+
+def _mean_ignore_negative(series):
+    """
+        Calculates the mean of the column without considering -1 entries in the column
+    """
+    return series[series != -1].mean()
```

Comment:
The name doesn't match the logic. Should the condition be `<0` or the name `ignore_minus_ones`?

Reply:
Updated the function name. -1 are intentionally ignored.